.circleci/check_gotest.sh

```diff
@@ -2,14 +2,29 @@
 
 # Run test in each go environment and log errors to $FILEPATH_ERROR_FLAG.
 # If all tests pass successfully then the file will not be created at all.
-export GOPATH=$HOME/$GOPATH_FOLDER;
 # cd to code with go.mod file outside of GOPATH to work with module-based behaviour
 # https://github.com/golang/go/wiki/Modules#when-do-i-get-old-behavior-vs-new-module-based-behavior
 cd $HOME/project
 
-for version in $VERSIONS; do
-    export GOROOT=$HOME/go_root_$version/go;
-    export PATH=$GOROOT/bin/:$PATH;
+OLD_PATH=$PATH
+
+if [ -z "$GO_VERSIONS" ]; then
+    GO_VERSIONS="$(echo $GOROOT | sed -E 's|^/usr/lib/go/([0-9]+\.[0-9]+(\.[0-9]+)?)/go$|\1|')"
+fi
```

Comment:
It would be nice to add a comment, like "test with all installed Go versions if no specific versions are requested" so that people won't need to decipher the regexp every time.

Reply:
Yeah, okay. Though, it's not "test with all installed Go versions", it's "test with Go version that is set by default in the container"